### DISCUSSION TITLE
how to define a default kong plugin configuration and it can be inherited  by other kong plugin objects

### DISCUSSION THREAD

**Discussion by edhikln** [Kong/kong#10277] [Help] [answered]
we are running enterprise kong with version 2.3.3.4 and kong ingress controller with version 0.9.0. rate-limit-advanced plugin will be used in our case. it will be integrated redis.
we want to define a default rate-limit-advanced plugin to keep redis configuration in kong deployment namespace. the actual rate-limit-advanced plugin defined by service component will inherit the redis configuration from default rate-limit-advanced plugin. the service component just focus on rate limit value configuration. they don't care where is redis. that means the redis configuration is only defined only place. it will not broadcast to the every service component.

how can i do that? thanks!

**Comment by kikito** [comment, marked as answer]
This is not possible at the moment. It might be possible in future versions of Kong, but however that is done it will not be backported to the version of the gateway that you have (2.3.3.4).

**Reply by edhikln** [reply]
thanks for your answers. we will find a way from deployment script perspective to avoid multiple redis configuration values.

**Comment by svyatoslavratov** [comment]
This is cool feature for flexible gitOps declarative config files
